src/Shared/E2ETesting/E2ETesting.targets

```diff
@@ -23,19 +19,46 @@
 
   <!-- Running prerequisites -->
 
-  <Target Name="EnsurePrerequisites" Condition="'$(EnforcePrerequisites)' == 'true'" BeforeTargets="EnsureNodeJSRestored">
+  <Target Name="EnsurePrerequisites">
+    <Message Condition="'$(EnforcePrerequisites)' == 'false'"
+      Importance="High"
+      Text="Prerequisites were not enforced at build time. Running Yarn or the E2E tests might fail as a result. Check /src/Shared/E2ETesting/Readme.md for instructions." />
+
     <PropertyGroup>
       <_PackageJson>$(MSBuildProjectDirectory)\package.json</_PackageJson>
     </PropertyGroup>
 
     <!-- JAVA -->
-    <!-- https://github.com/aspnet/AspNetCore-Internal/issues/2555 -->
     <Message Importance="High" Text="Ensuring JAVA is available" />
-    <Exec Command="java -version" />
-    <Message Importance="High" Text="JAVA is available on the PATH" />
+    <Exec Command="java -version" IgnoreExitCode="true">
+      <Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
+    </Exec>
+
+    <Message
+      Condition="'$(ErrorCode)' == '0'"
+      Importance="High"
+      Text="JAVA is available on the PATH" />
+    <Message
```

Comment:
Could consider making this a warning instead, applies in a few places.

Reply:
As I said above, get rid of the `'$(EnforcePrerequisites)' == 'false'` cases entirely. If a user passes `/p:EnforcePrerequisites=false` on the command line, assume they know Java is where it should be.